src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -349,8 +349,9 @@ export class CellModel extends Disposable implements ICellModel {
 	private attachImageFromSource(newSource: string | string[]): string | string[] {
 		if (!Array.isArray(newSource) && this.isValidBase64OctetStream(newSource)) {
 			let results;
-			while ((results = validBase64OctetStreamRegex.exec(newSource)) !== null) {
-				let imageName = this.addAttachment(results[1], results[0], 'image.png');
+			// only replace the base64 value if it's from markdown [](base64value) not html tags <img src="base64value"
+			while ((results = validBase64OctetStreamRegex.exec(newSource)) !== null && newSource.substring(newSource.indexOf(results[0]) - 9, newSource.indexOf(results[0]) - 2) !== 'img src') {
```

Comment:
Why not update the regex to only look at [](base64Value) values instead? That'd be cleaner and remove the chance of some other format breaking this.